src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -25,6 +25,648 @@
 )
 
 
+class ScanImageImagingExtractor(ImagingExtractor):
+    """
+    Specialized extractor for reading TIFF files produced via ScanImage software.
+
+    This extractor is designed to handle the structure of ScanImage TIFF files, which can contain
+    multi channel and multi volume data.  It also supports both single-file and multi-file datasets generated
+    by ScanImage in various acquisition modes (grab, focus, loop).
+
+    The extractor creates a mapping between each frame in the dataset and its corresponding physical file
+    and IFD (Image File Directory) location. This mapping enables efficient retrieval of specific frames
+    without loading the entire dataset into memory, making it suitable for large datasets.
+
+    Key features:
+    - Handles multi-channel data with channel selection
+    - Supports volumetric (multi-plane) imaging data
+    - Automatically detects and loads multi-file datasets based on ScanImage naming conventions
+    - Extracts and provides access to ScanImage metadata
+    - Efficiently retrieves frames using lazy loading
+
+    Current limitations:
+    - Does not support datasets with multiple frames per slice (will raise ValueError)
```

Comment:
Why can't this extractor deal with multiple frames per slice and/or flyback frames?

Reply:
I am adding it in a second PR because this thing is too large to review already.

Reply:
It is now ready on #417 but should come after this. Can you take a second look?

Reply:
Also https://github.com/catalystneuro/roiextractors/pull/419 for flyback frames.